jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt

```diff
@@ -159,7 +159,8 @@ class CodeWhispererPopupManager {
             return
         }
         val typeahead = resolveTypeahead(states, selectedIndex, typeaheadOriginal)
-        sessionContext = SessionContext(typeahead, typeaheadOriginal, selectedIndex, sessionContext.seen)
+        val isFirstTimeShowingPopup = indexChange == 0 && typeaheadChange.isEmpty()
```

Comment:
just for my understanding, could there be cases that it's first time showing popup but typeahead is not empty?

Reply:
It's not possible by design, the typeahead only tracks the input after some recommendation has already displayed in the editor. The thing that user types before the completion shows is called `userInputSinceInvocation` which is a different thing.